test/functional/discovered_hosts_controller_test.rb

```diff
@@ -180,7 +180,7 @@ def test_reboot_success_legacy
     host = discover_host_from_facts(facts)
     Host::Discovered::any_instance.stubs(:proxied?).returns(false)
     Host::Discovered::any_instance.stubs(:proxy_url).returns("http://1.2.3.4:8443")
-    ::ForemanDiscovery::NodeAPI::PowerLegacyDirectService.any_instance.expects(:reboot).returns(true)
+    ::ForemanDiscovery::NodeAPI::PowerService.any_instance.expects(:reboot).returns(true)
     ActiveSupport::Deprecation.silence do
       post "reboot", { :id => host.id }, set_session_user
     end
```

Comment:
There is also test_reboot_success_legacy test, can you remove that? It is useless I think. Other that that we can merge!